js/ai/src/prompt.ts

```diff
@@ -66,12 +66,8 @@ export function isPrompt(arg: any): boolean {
 export type PromptGenerateOptions<
   I = undefined,
   CustomOptions extends z.ZodTypeAny = z.ZodTypeAny,
-> = Omit<
-  GenerateOptions<z.ZodTypeAny, CustomOptions>,
-  'prompt' | 'input' | 'model'
-> & {
+> = Omit<GenerateOptions<z.ZodTypeAny, CustomOptions>, 'prompt' | 'model'> & {
```

Comment:
`model` is already optional in `GenerateOptions`, why omit it just to put it back? Also, I think we need to get rid of `I` and replace it with `O` like this and carry it through in the functions:

```
export type PromptGenerateOptions<
  O extends z.ZodTypeAny = z.ZodTypeAny,
  CustomOptions extends z.ZodTypeAny = z.ZodTypeAny,
> = Omit<GenerateOptions<O, CustomOptions>, 'prompt'>;
```

Reply:
updated

Reply:
?

Reply:
I mean updated now...